drivers/android/defs.rs

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: GPL-2.0
+
+use core::ops::{Deref, DerefMut};
+use kernel::{
+    bindings,
+    user_ptr::{ReadableFromBytes, WritableToBytes},
+};
+
+pub const BR_ERROR: u32 = bindings::binder_driver_return_protocol_BR_ERROR;
+pub const BR_OK: u32 = bindings::binder_driver_return_protocol_BR_OK;
```

Comment:
Perhaps a macro to generate all of these easily? i.e.

    generate_br_constants!(TRANSACTION, REPLY, DEAD_REPLY ......)

Reply:
The problem is that given a name, say, `BR_TRANSACTION`, we need to generate a new identifier `bindings::binder_driver_return_protocol_BR_TRANSACTION`, which is a concatenation. It looks like we could use `concat_idents!` but it's unstable.

I'd rather leave this as is for now and focus on the other missing features.

Reply:
Of course, don't worry about it. I was thinking more from the PoV of readers of the RFC, thinking "Rust cannot do this with a macro??" (because the "block" of definitions stands out when scrolling).

Reply:
I see. I understand the concern now. Let me see if I can do it with a macro.

Reply:
Unfortunately, it looks like we need proc macros to do this, from https://doc.rust-lang.org/std/macro.concat_idents.html:
> That means while you may use this macro for referring to existing variables, functions or modules etc, you cannot define a new one with it.

We can either write one or import some crate that does this for us. Both options are undesirable.

Do you see any other way to improve this?

Reply:
Bummer... Nah, I wouldn't spend more time on this. Perhaps just add `TODO` comment so that people now when reading.

Reply:
We could always run the C preprocessor on Rust files, though. 😈

Reply:
We actually only _refer_ to concatenated names, it may be possible to use macros. Let me try again.

Reply:
If you put the `BR_` in the arguments, yeah, it would only be a refer, indeed. Good catch.

Reply:
We needed to include `BR_` anyway because we also have `BC_`. Also needed to import `bindings::*` but it works!

Reply:
Looks nice :)